src/lambda.ts

```diff
@@ -48,7 +55,9 @@ export class WatchLambdaFunction extends Construct {
 
     const cfnFunction = props.fn.node.defaultChild as lambda.CfnFunction;
     // if we have an IFunction or an unspecified timeout, use the default of 3 seconds
-    const timeoutSec = cfnFunction? (cfnFunction.timeout? cfnFunction.timeout : 3) : 3;
+    let timeoutSec = cfnFunction? (cfnFunction.timeout? cfnFunction.timeout : 3) : 3;
+    // if a threshold was passed for duration use it
+    timeoutSec = props.durationTimeout? props.durationTimeout: timeoutSec;
```

Comment:
I'm not 100% on this syntax, but I think it could just be

```suggestion
    const timeoutSec = props.durationTimeout || cfnFunction.timeout || 3;
```

Reply:
unfortunately, if the lambda wasn't created with cdk, we can't even use the `cfnFunction.timeout` expression if it isn't after a `cfnFunction?` clause